### PR TITLE
[thread.lock] Extract error conditions from 'Throws' element.

### LINKED ISSUE #458: [thread.lock.unique.locking] and [thread.lock.shared.locking] need more bullets
The definitions of `lock`, `try_lock`, `try_lock_for` and `try_lock_until` just list exceptions under the _Throws_ elements, rather than with a bulleted _Error conditions_ element like the rest of the clause.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #458.

### CODE REVIEW

File: source/threads.tex [hunk 1]
```diff
@@ -2462,7 +2494,15 @@
 \postconditions \tcode{owns == res}, where \tcode{res} is the value returned by the call to \tcode{pm->try_lock_shared_for(rel_time)}.
 
 \pnum
-\throws Any exception thrown by \tcode{pm->try_lock_shared_for(rel_time)}. \tcode{system_error} if an exception is required ~(\ref{thread.req.exception}). \tcode{system_error} with an error condition of \tcode{operation_not_permitted} if \tcode{pm} is \tcode{nullptr}. \tcode{system_error} with an error condition of \tcode{resource_deadlock_would_occur} if on entry \tcode{owns} is \tcode{true}.
+\throws Any exception thrown by \tcode{pm->try_lock_shared_for(rel_time)}. \tcode{system_error} when an exception is required ~(\ref{thread.req.exception}).
```
Comment (zygoloid) [inline source/threads.tex]: Please remove the space in `required ~(\ref{` here.

File: source/threads.tex [hunk 2]
```diff
@@ -1968,11 +1968,16 @@
...
-is required~(\ref{thread.req.exception}). \tcode{system_error} with an error
-condition of \tcode{operation_not_permitted} if \tcode{pm} is 0. \tcode{system_error}
-with an error condition of \tcode{resource_deadlock_would_occur} if on entry \tcode{owns}
+Any exception thrown by \tcode{pm->lock()}. \tcode{system_error} when an exception
+is required~(\ref{thread.req.exception}).
+
+\pnum
+\errors
+\begin{itemize}
+\item \tcode{operation_not_permitted} --- if \tcode{pm} is 0.
```
Comment (zygoloid) [inline source/threads.tex]: Can we consistently use `nullptr` instead of `0` in these conditions?

Comment (jensmaurer) [inline source/threads.tex]: Sure, but isn't that a different issue / different patch?

**Comment by jensmaurer** [comment]
Rebased + issues fixed (including using `nullptr`).

**Pull request merged**